discovery/pkg/sync/endpoints.go

```diff
@@ -54,63 +55,71 @@ func (action endpointsAction) ObjectMeta() *metav1.ObjectMeta {
 }
 
 // Sync performs the action on the given Endpoints resource
-func (action endpointsAction) Sync(kubeClient kubernetes.Interface, metrics localmetrics.DiscovererMetrics, backendName string) error {
+func (action endpointsAction) Sync(kubeClient kubernetes.Interface, metrics localmetrics.DiscovererMetrics, logger *logrus.Logger) error {
 
 	var err error
 	switch action.kind {
 	case actionAdd:
-		err = addEndpoints(kubeClient, action.endpoints, metrics, backendName)
+		err = addEndpoints(kubeClient, action.endpoints, metrics)
 	case actionUpdate:
-		err = updateEndpoints(kubeClient, action.endpoints, metrics, backendName)
+		err = updateEndpoints(kubeClient, action.endpoints, metrics)
 	case actionDelete:
-		err = deleteEndpoints(kubeClient, action.endpoints, metrics, backendName)
+		err = deleteEndpoints(kubeClient, action.endpoints, metrics)
 	}
 	if err != nil {
 		return fmt.Errorf("error handling %s: %v", action, err)
 	}
 
-	metrics.EndpointsEventTimestampMetric(action.endpoints.GetNamespace(), backendName, action.endpoints.GetName(), time.Now().Unix())
+	metrics.EndpointsEventTimestampMetric(action.endpoints.GetNamespace(), action.endpoints.GetName(), time.Now().Unix())
+
+	// TODO: Move to lister()
+	totalEps, err := metrics.GetTotalEndpointsCount(kubeClient, action.endpoints.GetNamespace(), action.endpoints.GetName())
```

Comment:
I am not following why we need to GET the endpoints object again, if we already have it in `action.endpoints`?

Reply:
Good point, before I had the metric sum all endpoints for a namespace which was different.